connectors/services/sync.py

```diff
@@ -164,3 +177,32 @@ async def _run(self):
                 await self.sync_job_index.close()
             await es.close()
         return 0
+
+    async def _ready_to_sync(self, connector, data_provider):
+        try:
+            next_sync = connector.next_sync()
+            # First we check if sync is disabled, and it terminates all other conditions
+            if next_sync == SYNC_DISABLED:
+                logger.debug(f"Scheduling is disabled for connector {connector.id}")
+                return False
+            # Then we check if we need to restart SUSPENDED job
+            if connector.last_sync_status == JobStatus.SUSPENDED:
+                logger.debug("Restarting sync after suspension")
+                return True
+            # And only then we check if we need to run sync right now or not
+            if next_sync - self.idling > 0:
+                logger.debug(
+                    f"Next sync for connector {connector.id} due in {int(next_sync)} seconds"
+                )
+                return False
+            # Lastly we skip job scheduling if there's no change in the remote source
+            if not await data_provider.changed():
```

Comment:
Hi @tarekziade , in the current implementation, when customers click `Sync` in Kibana or scheduled a sync, but there's no change in the remote source, there won't be any job scheduled. Especially for the forced sync, it will definitely confuse customers and they may think this is a bug. cc @danajuratoni 

Another approach is that we skip this check and schedule a job, and directly mark the job as `completed` during job execution:
https://github.com/elastic/connectors-python/blob/c71b575df81846e0ac4976e5b1278cc53155fdd9/connectors/sync_job_runner.py#L66-L71

WDYT?

Reply:
Makes a lot of sense